app/components/publications/batch-documents-publication-modal.js

```diff
@@ -104,15 +106,52 @@ export default class PublicationsBatchDocumentsPublicationModalComponent extends
 
   @action
   async linkPublicationFlow(piece, publicationFlow) {
+    let publicationFlowHasChanged = false;
+
+    const _case = await publicationFlow.case;
+    if (this.case.id != _case.id) {
+      const agendaItemTreatment = await publicationFlow.agendaItemTreatment;
+      const agendaitem = await this.store.queryOne('agendaitem', {
+        'filter[treatments][:id:]': agendaItemTreatment.id
+      });
+      if (!agendaitem) {
+        // Selected publication-flow is currently linked to another case (not via MR).
+        // We need to relink the publication-flow to this case and cleanup the dummy data
+        // that was created for the publication-flow back then.
+        publicationFlow.case = this.case;
+        publicationFlow.agendaItemTreatment = this.agendaItemTreatment;
+        publicationFlowHasChanged = true;
+        await Promise.all([
+          _case.destroyRecord(),
+          agendaItemTreatment.destroyRecord(),
+        ]);
+      } else {
+        // Publication-flow is already linked to an agenda-item-treatment that has been
+        // handled on an agenda. We cannot relink the publication-flow in that case.
+        // In practice, this scenario will not occur since they are not able to select
+        // such a publication-flow in the publication-flow-selector.
+        this.toaster.error(
+          this.intl.t('unable-to-relink-publication-flow'),
+          this.intl.t('warning-title')
+        );
+        throw Error('Unable to relink publication flow');
```

Comment:
Nice catch. It was throwing an error because I didn't expect to hit this case. Updated the inline documentation with the concurrency scenario and replaced the error-throwing with a regular return to exit the method :+1: